### PR TITLE
Milestone 1 :Creating data model

### PR DESCRIPTION
**Pull request by ibrahim777764**
### Milestone 1 :Creating data model

- Install Ruby on Rails framework.
- Used Ruby Gems as a software packages system.
- Understood what ORM is.
- Used database migration files to maintain database schema.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,36 @@
...
-Things you may want to cover:
+- ![](https://img.shields.io/badge/Github-blueviolet)
+- ![](https://img.shields.io/badge/Ruby-red)
+- ![](https://img.shields.io/badge/Ruby*on*Rails-red)
 
-* Ruby version
+## Authors
 
-* System dependencies
+👤 **ibrahim naseer**
```
Comment (dasileker) [inline README.md]: - [ ] Hello and good work on the app, but please add relevant instructions on how to clone and run the app.

File: db/migrate/20220920110226_add_user_ref_to_post.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddUserRefToPost < ActiveRecord::Migration[7.0]
+  def change
```
Comment (dasileker) [inline db/migrate/20220920110226_add_user_ref_to_post.rb]: - [ ] Please follow the ERD provided for this tutorial, right now you are not affecting the Author as a reference for the user.

File: db/migrate/20220920110243_add_user_ref_to_like.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddUserRefToLike < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :likes, :user, null: false, foreign_key: true
+  end
```
Comment (dasileker) [inline db/migrate/20220920110243_add_user_ref_to_like.rb]: - [ ] Please follow the ERD provided for this tutorial, right now you are not affecting the Author as a reference for the user.

File: db/migrate/20220920110257_add_post_ref_to_likes.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddPostRefToLikes < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :likes, :post, null: false, foreign_key: true
+  end
+end
```
Comment (dasileker) [inline db/migrate/20220920110257_add_post_ref_to_likes.rb]: - [ ] Please follow the ERD provided for this tutorial, right now you are not affecting the Author as a reference for the user.

File: db/migrate/20220920110309_add_post_ref_to_comment.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddPostRefToComment < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :comments, :post, null: false, foreign_key: true
+  end
+end
```
Comment (dasileker) [inline db/migrate/20220920110309_add_post_ref_to_comment.rb]: - [ ] Please follow the ERD provided for this tutorial, right now you are not affecting the Author as a reference for the user.

File: db/migrate/20220920110211_add_user_ref_to_comment.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddUserRefToComment < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :comments, :user, null: false, foreign_key: true
+  end
```
Comment (dasileker) [inline db/migrate/20220920110211_add_user_ref_to_comment.rb]: - [ ] Please follow the ERD provided for this tutorial, right now you are not affecting the Author as a reference for the user.

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @ibrahim777764,** 

### **Good job so far!**
There are some issues that you still need to work on to go to the next project but you are almost there!

## **Highlights**
- Lintesr are passing ✅.
- PR is well made 👌.
- Commits are consistent 👌.


## Required Changes ♻️
_Check the comments under the review._
- Hello and excellent work on the app, but there are some migration files that are empty please remove them.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — APPROVED** [review]
Hi Ibrahim, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- good job following the instructions from the previous CR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**